service/history/handler/handler.go

```diff
@@ -1601,6 +1601,9 @@ func (h *handlerImpl) GetReplicationMessages(
 
 		size := proto.FromReplicationMessages(tasks).Size()
 		if (responseSize + size) >= maxResponseSize {
+			metricsScope.Tagged(metrics.ShardIDTag(int(shardID)))
```

Comment:
it would be better to have a gauge metric instead of counter which would have value `responseSize + size`

Reply:
I see. If we are more interested in the number of occurrences then counter makes sense.